### ISSUE TITLE
False error on redundant cast for super globals

### ISSUE THREAD

**Issue by franck-paul** [vimeo/psalm#11554]
Example:

```php
$uri_parts = explode('?', (string) $_SERVER['REQUEST_URI']);
````

Give me `ERROR: RedundantCast - ... - Redundant cast to string (see https://psalm.dev/262)` but it should not as `$_SERVER` is `array<mixed>`.

**Comment by franck-paul** [comment]
See https://psalm.dev/r/1a6d91ff45

**Comment by orklah** [comment]
$_SERVER['REQUEST_URI'] should always contain a string (unless you somehow rewrote the value but then, Psalm won't help you!)
